oak_restricted_kernel/src/mm/mod.rs

```diff
@@ -17,25 +17,42 @@
 use goblin::{elf32::program_header::PT_LOAD, elf64::program_header::ProgramHeader};
 use log::info;
 use oak_linux_boot_params::{BootE820Entry, E820EntryType};
+use sev_guest::msr::{get_sev_status, SevStatus};
 use x86_64::{
     addr::{align_down, align_up},
     registers::{
         control::{Cr3, Cr3Flags},
         model_specific::{Efer, EferFlags},
     },
     structures::paging::{
-        FrameAllocator, OffsetPageTable, PageSize, PageTable, PageTableFlags, PhysFrame, Size2MiB,
-        Size4KiB,
+        FrameAllocator, MappedPageTable, Page, PageSize, PageTable, PhysFrame, Size2MiB, Size4KiB,
     },
     PhysAddr, VirtAddr,
 };
 
-use self::page_tables::DirectMap;
+use self::encrypted_mapper::{EncryptedPageTable, MemoryEncryption, PageTableFlags, PhysOffset};
 
 mod bitmap_frame_allocator;
+mod encrypted_mapper;
 pub mod frame_allocator;
 pub mod page_tables;
 
+pub trait Translate {
+    /// Translates the given virtual address to the physical address that it maps to.
+    ///
+    /// Returns `None` if there is no valid mapping for the given address.
+    fn translate(&self, addr: VirtAddr) -> Option<PhysAddr>;
+
+    /// Translate a physical address to a virtual address.
+    ///
+    /// Note that a physical address may be mapped multiple times. This function will always return
+    /// the address from the directly mapped region, ignoring ohter mappings if they exist.
+    fn translate_addr(&self, addr: PhysAddr) -> Option<VirtAddr>;
```

Comment:
The method names do not seem particularly related to the actual functionality. e.g. it's unclear why `translate` and `translate_addr` (just looking at their names) operate in opposite directions. Maybe they could be called `virtual_to_physical` and `physical_to_virtual` or something like that?

Reply:
It was mostly due to function names in the existing `x86_64` `Translate` trait. But yes, I agree, the names were confusing, so I've renamed them to something hopefully more readable.